File: 프로그래머스/3/43162. 네트워크/네트워크.py
import sys
sys.setrecursionlimit(10**6)

def solution(n, computers):
    answer = 0
    visited = [0] * n
    def dfs(i):
        visited[i] = True
        for j in range(n):
            if computers[i][j] == 1 and not visited[j]:
                dfs(j)
    for i in range(n):
        if not visited[i]:
            dfs(i)
            answer += 1
    
    return answer

File: 프로그래머스/3/43162. 네트워크/네트워크.java
class Solution {

    static int N;
    static int[][] coms;
    static boolean[] visited;

    private void dfs(int x) {
        visited[x] = true;
        for (int i = 0; i < N; i++) {
            if (coms[x][i] == 1 && !visited[i]) {
                dfs(i);
            }
        }
    }

    public int solution(int n, int[][] computers) {
        int answer = 0;
        N = n;
        coms = computers;
        visited = new boolean[N];

        for (int i = 0; i < N; i++) {
            if (!visited[i]) {
                dfs(i);
                answer++;
            }
        }

        return answer;
    }
}